testinfra/utils/ansible_runner.py

```diff
@@ -18,150 +18,183 @@
 from __future__ import unicode_literals
 from __future__ import absolute_import
 
-import pprint
+import datetime
+import json
+import os
+import shutil
+import subprocess
+import tempfile
+import yaml
 
 
-try:
-    import ansible
-except ImportError:
-    raise RuntimeError(
-        "You must install ansible package to use the ansible backend")
+# the real ansible-runner
+import ansible_runner
 
-import ansible.cli.playbook
-import ansible.constants
-import ansible.executor.task_queue_manager
-import ansible.inventory
-import ansible.parsing.dataloader
-import ansible.playbook.play
-import ansible.plugins.callback
-import ansible.utils.vars
-import ansible.vars
 
-try:
-    from ansible.module_utils._text import to_bytes
-except ImportError:
-    from ansible.utils.unicode import to_bytes
+__all__ = ['AnsibleRunner', 'to_bytes']
 
 
-__all__ = ['AnsibleRunner', 'to_bytes']
+def to_bytes(data):
+    '''Why!?!?'''
+    return b'%s' % data
 
 
-class AnsibleRunnerBase(object):
-    _runners = {}
+class AnsibleInventoryException(Exception):
+    def __init__(self, message):
+        super(AnsibleInventoryException, self).__init__(message)
+        self.message = message
 
-    def __init__(self, host_list=None):
-        self.host_list = host_list
-        super(AnsibleRunnerBase, self).__init__()
 
-    def get_hosts(self, pattern=None):
-        raise NotImplementedError
+class AnsibleRunnerV2(object):
 
-    def get_variables(self, host):
-        raise NotImplementedError
+    # testinfra api
+    _runners = {}
 
-    def run(self, host, module_name, module_args, **kwargs):
-        raise NotImplementedError
+    # testinfra api
+    host_list = None
+
+    # variable cache
+    variables = {}
+
+    def __init__(self, host_list=None):
+        # host_list is the list of inventory files, aka -i
+        self.host_list = host_list
 
     @classmethod
     def get_runner(cls, inventory):
-        try:
-            return cls._runners[inventory]
-        except KeyError:
+        # stores a copy of the runner in a dict keyed by inv
+        if inventory not in cls._runners:
             cls._runners[inventory] = cls(inventory)
-            return cls._runners[inventory]
+        return cls._runners[inventory]
+
+    def fetch_inventory(self, host=None):
+        '''Helper function for ansible-inventory'''
+
+        cmd = 'ansible-inventory -i %s' % self.host_list
+        if host is not None:
+            cmd += ' --host=%s' % host
+        else:
+            cmd += ' --list'
+
+        env = os.environ.copy()
+        env['ANSIBLE_NOCOLOR'] = "1"
+        p = subprocess.Popen(
+            cmd,
+            shell=True,
```

Comment:
I'd use ``subprocess.check_output` here and pass cmd as a list to avoid shell injections.